dart/example/monster_my_game.sample_generated.dart

```diff
@@ -0,0 +1,456 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+// ignore_for_file: unused_import, non_constant_identifier_names
+
+library my_game.sample;
+
+import 'dart:typed_data' show Uint8List;
+import 'package:flat_buffers/flat_buffers.dart' as fb;
+
+
+class Color {
```

Comment:
Is this the only way to do enums in Dart? Can't we have integer constants instead somehow (without a class)?

Reply:
Raw enums in dart exists but must be 0 indexed, which is not the case in FlatBuffers.  If we used raw integer constants we'd lose type-safety which doesn't seem like a good tradeoff in this case, but could be possible.  My preference would be to keep the enum-like classes here.

Reply:
It is more important that it matches the semantics of FlatBuffers I feel, which represents them as integers. If enums cause object allocation, that be quite odd. The other languages also prefer speed over convenience whereever applicable.

If you feel strongly about convenience, a seperate "object API" (that sits on top of the base API) can be added.. one exists for C++ and is planned for other languages.

Reply:
I think it'd make sense to refactor this then to use raw ints and have an enum helper class available with some kind of `toInt`/`fromInt` methods/constructors.  I'll work on that.

Reply:
Actually, reexamining the current implementation - it isn't doing all the allocating it seems to at first.  The use of `const` on these means they're all only allocated once: 
```
  static const Color Red = const Color._(0);
  static const Color Green = const Color._(1);
  static const Color Blue = const Color._(2);
```
These only require the initial allocations for each value - so three allocations.  Those three allocations are statically stored in the program and reused - there's no 'new' allocation used anywhere related to these (it's always `const`).  A change to this to use ints would be fairly simple, but it would also be less clear and not really save a whole lot in the end.  Are you still opposed to having it done this way?

Reply:
I guess that is much more acceptable then. If this is idiomatic for doing enums in Dart I guess it is ok.

My thinking is that you should want a representation that matches FlatBuffers semantics (which is cross-language) rather than trying to force the language way of doing things.